server/src/test-integration/java/com/thoughtworks/go/server/web/UrlRewriterIntegrationTest.java

```diff
@@ -43,23 +44,139 @@
+    @DataPoint
+    public static ResponseAssertion ARTIFACT_API_GET_FILE = new ResponseAssertion(HTTP_URL + "/go/files/pipeline/1/stage/1/job//tmp/file", HTTP_URL + "/go/repository/restful/artifact/GET/?pipelineName=pipeline&pipelineLabel=1&stageName=stage&stageCounter=1&buildName=job&filePath=%2Ftmp%2Ffile", true);
+    @DataPoint
+    public static ResponseAssertion ARTIFACT_API_PUSH_FILE = new ResponseAssertion(HTTP_URL + "/go/files/pipeline/1/stage/1/job//tmp/file", HTTP_URL + "/go/repository/restful/artifact/POST/?pipelineName=pipeline&pipelineLabel=1&stageName=stage&stageCounter=1&buildName=job&filePath=%2Ftmp%2Ffile", METHOD.POST, true);
+    @DataPoint
+    public static ResponseAssertion ARTIFACT_API_CHANGE_FILE = new ResponseAssertion(HTTP_URL + "/go/files/pipeline/1/stage/1/job/file", HTTP_URL + "/go/repository/restful/artifact/PUT/?pipelineName=pipeline&pipelineLabel=1&stageName=stage&stageCounter=1&buildName=job&filePath=file", METHOD.PUT, true);
+
+    @DataPoint
+    public static ResponseAssertion PIPELINE_DASHBOARD_JSON = new ResponseAssertion(HTTP_URL + "/go/pipelines.json", HTTP_URL + "/go/rails/pipelines.json", METHOD.GET);
+    @DataPoint
+    public static ResponseAssertion MATERIALS_VALUE_STREAM_MAP = new ResponseAssertion(HTTP_URL + "/go/materials/value_stream_map/fingerprint/revision", HTTP_URL + "/go/rails/materials/value_stream_map/fingerprint/revision", METHOD.GET);
+
+    @DataPoint
+    public static ResponseAssertion RAILS_INTERNAL_API = new ResponseAssertion(HTTP_URL + "/go/api/config/internal/pluggable_task/indix.s3fetch", HTTP_URL + "/go/rails/api/config/internal/pluggable_task/indix.s3fetch", METHOD.GET);
+
+    @DataPoint
+    public static ResponseAssertion USERS_INDEX_API = new ResponseAssertion(HTTP_URL + "/go/api/users", HTTP_URL + "/go/rails/api/users", METHOD.GET);
+    @DataPoint
+    public static ResponseAssertion USERS_SHOW_API = new ResponseAssertion(HTTP_URL + "/go/api/users/some.one", HTTP_URL + "/go/rails/api/users/some.one", METHOD.GET);
+
+    @DataPoint
+    public static ResponseAssertion DATA_SHARING_SETTINGS = new ResponseAssertion(HTTP_URL + "/go/api/data_sharing/settings", HTTP_URL + "/go/spark/api/data_sharing/settings", true);
+
+    @DataPoint
+    public static ResponseAssertion DATA_SHARING_USAGE_DATA = new ResponseAssertion(HTTP_URL + "/go/api/internal/data_sharing/usagedata", HTTP_URL + "/go/spark/api/internal/data_sharing/usagedata", true);
+
+    @DataPoint
+    public static ResponseAssertion DATA_SHARING_REPORTING = new ResponseAssertion(HTTP_URL + "/go/api/internal/data_sharing/reporting", HTTP_URL + "/go/spark/api/internal/data_sharing/reporting", true);
+
+    @DataPoint
+    public static ResponseAssertion LANDING_PAGE_SLASH = new ResponseAssertion(HTTP_URL + "/go/", HTTP_URL + "/go/spark/dashboard", true);
+
+    @DataPoint
+    public static ResponseAssertion LANDING_PAGE_HOME = new ResponseAssertion(HTTP_URL + "/go/home", HTTP_URL + "/go/spark/dashboard", true);
```

Comment:
Add following data points

```java
@DataPoint
public static ResponseAssertion LANDING_PAGE_HOME_SLASH = new ResponseAssertion(HTTP_URL + "/gO/hOme/", HTTP_URL + "/go/spark/dashboard", true);
    
@DataPoint
public static ResponseAssertion LANDING_PAGE_GO = new ResponseAssertion(HTTP_URL + "/gO", HTTP_URL + "/go/spark/dashboard", true);
```

Reply:
@bdpiparva I mentioned this [here](https://github.com/gocd/gocd/pull/4874#discussion_r197290445). I don't this it is important. Do you know why `/Go` etc. should exist? I've never seen it be used that way.

Reply:
I know it is not used anywhere in code-base. This is just to make sure that it should not return 404 in case if someone has bookmarked their GoCD server URL till `/go`.

Reply:
With this PR if you start GoCD server and try to access `http://localhost:8253/gO/` will return `404` earlier which was redirected to `/go/pipelines`

Reply:
@bdpiparva Ok. Unfortunately, `ContextHandlerCollection` (and servlet-api) seem to provide no way to handle this case insensitively. These work already: `/go/HOME`, `/go/Home`. However, each handler can only claim one case-sensitive context path. We will have to live with `/gO` or `/GO` breaking, imo. Unless you have any other ideas.